---
title: "Figures_Blob_Project_v30"
author: "S. J. Traving, T. T. E. Kellogg, T. Ross, R. McLaughlin, B. Kieft, G. Y. Ho, M. Robert, S. J. Hallam"
date: '2020-07-22'
output: html_document
---

This is the Rmarkdown documentation for all figures and supplementary figures, unless otherwise described, for the manuscript "Microbial responses to a warm temperature anomaly in the Northeast subarctic Pacific Ocean".

## 16S amplicon data
Three biological samples were removed due to an error of mislabeled during processing (samples P26.2011.02.1500m, P26.2011.02.3000m, P26.2011.02.4000m).

OTUs with abundances < 2 reads were removed from the OTU table. Furthermore, OTU_164, _726 and _1176 were removed as they failed taxonomy assignment in QIIME2 (Bolyen et al., 2019,  https://doi.org/10.1038/s41587-019-0209-9) and a manual blast search yielded no better result (NCBI, https://www.ncbi.nlm.nih.gov/).

Set working directory:

```{r setup, cache.comments=FALSE, warning=FALSE, echo=FALSE}
setwd("/Users/brandonkieft/Downloads/markdown/Blob_figures/")
```

## Colors
Creating color schemes used for the different figures:
```{r, echo = FALSE}

# Colors for data being grouped by Blob vs normal years
Blob.colors = c("#011F65", # Cool Black (actually blue
                "#EE452A" #Cinnabar
                )
# Color scheme for all depths
DepthCol2 =c(
  "#cc0000",
  "#E58621",
  "#FABB1C",
  "#908612",
  "#000080",
  "#8EAE04",
  "#286C00",
  "#00BC51",
  "#0FFFB7",
  "#00958A",
  "#00BCD6",
  "#99D9EA",
  "#0091FB",
  "#816FFF",
  "#b68dfa",
  "#D21E8E",
  "#BB719F",
  "#E2A8FF"
)

# Colors assigned directly
DepthColSET =c(
  "10" = "#cc0000",
  "25" = "#E58621",
 "50" = "#FABB1C",
 "100" = "#908612",
  "125" = "#000080",
 "150" = "#8EAE04",
  "200" = "#286C00",
 "300" = "#00BC51",
  "400" = "#0FFFB7",
  "600" = "#00958A",
  "800"= "#00BCD6",
 "1000"= "#99D9EA",
 "1250" = "#0091FB",
  "1500" = "#816FFF",
 "2000" = "#b68dfa",
 "3000" = "#D21E8E",
 "4000" = "#BB719F",
 "B-10" = "#E2A8FF")
```

## Preprocessing of OTU table and metadata

### Phyloseq
The R package Phyloseq by McMurdie and Holmes (2013), https://doi.org/10.1371/journal.pone.0061217 was used for different figures with data aggregated at different levels of taxonomic resolution. See also figures 3, 4 and supplementary figures 3, 5

Last updated: 16-08-2019
Creating Phyloseq object from OTU table for later use:

```{r, cache.comments = FALSE, warning = FALSE, echo = FALSE}
library(phyloseq) # v. 1.22.3
library(tidyverse) # v. 1.2.1

## raw OTU table with chloroplasts and mitochondrial reads removed
otu_table_data_frame = read.csv("otu_table_SILVA132_tax_noCM.csv", header =T , row.names=1) 
## copy over row names as separate column
otu_table_t <- rownames_to_column(as.data.frame(t(otu_table_data_frame)), var = "OTUID") 
tax_table_matrix = as.matrix(read.csv("otus_tax_SILVA132_noCM.csv", header=T, row.names=1)) 
## Taxonomy data with no chloroplast and mitochondrial data
meta_dat = read.csv("metadata_phyloseq5.csv", header=T, row.names=1)

## creating the Phyloseq components
otu_table_notax = otu_table(otu_table_data_frame[,1:271], taxa_are_rows = TRUE)
tax_table_phyloseq = tax_table(tax_table_matrix)
metadata = sample_data(meta_dat)

## Create Phyloseq object
P26 = merge_phyloseq(otu_table_notax, tax_table_phyloseq, metadata)
```

## Data transformations

### VST Transformation of OTU table
To account for data structure and distribution different normalizations or transformation methods are applied to the OTU table, depending on the best method for a particular analysis downstream.
The variance stabilized transformation (VST) from the DESEq2 package (Love et al., 2014, doi: 10.1186/s13059-014-0550-8) is one of the transformation methods used. 
VST as implemented in DESeq2 is equivalent to a log transformation regarding high abundant OTUs and the two transformations yield the same significant OTUs. However, for the low and mid range abundances the VST performs better results, due to VST taking size factors into account.
The default in calculating VarianceStabilizingTransformation, blind is default "TRUE".

Last updated: 16-08-2018
```{r, echo = FALSE}
library(DESeq2) # v. 1.18.1
library(S4Vectors)
library(stats4)
library(BiocGenerics)
library(parallel)

otu_table_data_frame = read.csv("otu_table_SILVA132_tax_noCM.csv", header =T , row.names=1)
dds <- otu_table_data_frame[,1:271]
meta_dat1 = read.csv("metadata_phyloseq5.csv", header=T) # Metadata with Blob category
metadata <- as.data.frame(meta_dat1)
dds1 <- DESeqDataSetFromMatrix(countData = dds, colData = metadata, design = ~ Depths)

# Setting type to "iterate" when estimating size factors didnt work, therefore the manual geoMeans
cts <- counts(dds1)
geoMeans <- apply(cts,1,function(row) if (all(row==0)) 0 else
  exp(sum(log(row[row !=0]))/length(row)))

dds1 <- estimateSizeFactors(dds1, geoMeans = geoMeans)
dds1 <- estimateDispersions(dds1)
dds2 <- varianceStabilizingTransformation(dds1, blind = TRUE)

write.table(assay(dds2), file="varianceStabilizingTransformation.txt", sep="\t")
vsd2 <- read.delim("VarianceStabilizingTransformation.txt", sep = "\t")

colnames(vsd2) <- dds1$SampleID

write.table(vsd2, file="vst_otu_table.txt", sep="\t")
write.table(vsd2, file="vst_otu_table.csv", sep=",")
write.table(t(vsd2), file="vst_otu_table_transposed.csv", sep=",")
# Correct header alignments
```
Stitch together VST matrix with taxonomic information for downstream analyses:
```{r, echo =FALSE}
tax<- read.csv("otu_table_SILVA132_tax_noCM.csv", header = TRUE)
tax <- tax[,c(1,273)]
vst <- read.delim("vst_otu_table.txt", header = TRUE)
vst$OTUID = rownames(vst)
vst.tax <- inner_join(vst, tax, by = c("OTUID" = "OTUID"))
write.table(vst.tax, file = "vst_otu_table_tax.txt", sep = "\t")
# Correct header alignment
```

### Proportions

The OTU table was also transformed to proportional abundance, as the VST performs poorly in a indicator species analysis (Hahn, 2016 phd thesis) which is applied downstreams.
```{r, echo = FALSE}
# Transform data to proportions
# Use Phyloseq object from above.
otu.prop = transform_sample_counts(P26, function(x) x / sum(x)) #proportions
write.csv(t(otu_table(otu.prop)), file("proportions_otu_table_trans.csv"))
otu.propt <- read.csv("proportions_otu_table_trans.csv", header =TRUE)
colnames(otu.propt)[1] = "SampleID"

# Proportional abundance matrix is joined with the metadata 
all.dat <- inner_join(meta_dat1,otu.propt, by = 'SampleID')
# save for future use
write.csv(all.dat, file("otu_prop_w_metadata_phyloseq5.csv"))

```

### Pigment data
Pigment data was provided by the Institute of Ocean Sciences (DFO/IOS), Sidney, Canada.
pigment samples are collected in duplicates, using an ethanol extraction and measured using HPLC, SOP by IOS.
Because sample depth varied between cruises the pigment data was binned in order to compare with amplicon sample depths.

pigment bins:   amplicon sample depths:
0-15m           10m
16-30m          25m
31-60m          50m

```{r, echo=FALSE}
library(tidyverse)
pigment <- read.csv("./Phyto_OSP_Jun2010toFeb2016_v2.csv", header = TRUE, stringsAsFactors = FALSE)
#average depth categories
pigment.avg <- pigment %>%
  group_by(Cruisetrans, Cat_Depth) %>%
  summarise(Cyanobacteria_avg = mean(Cyanobacteria.mg.m3),
            Chlorophytes_avg = mean(Chlorophytes.mg.m3),
            Prasinophytes_avg = mean(Prasinophytes.mg.m3),
            Cryptophytes_avg = mean(Cryptophytes.mg.m3),
            Diatom2_avg = mean(Diatom_2.mg.m3),
            Dinoflage1_avg = mean(Dinoflage_1.mg.m3),
            Pelagophytes_avg = mean(Pelagophytes.mg.m3),
            Haptophytes_avg = mean(Haptophytes.mg.m3),
            Chlorophyll_avg = mean(Chlorophyll.mg.m3)) 

meta_dat1 <- tibble::rownames_to_column(meta_dat, "VALUE") %>%
  left_join(pigment.avg, by = c("VALUE" = "Cruisetrans"))
write.table(meta_dat1,"metadata_with_pigments.tsv",sep="\t",row.names = F)
```

# Analyses and figures

General comment:
When data is presented at the Phyla level, Proteobacteria is always broken down to Class level, unless otherwise stated.

### Indicator analysis
The R package indicspecies by De Caceres and Legendre (2009), https://doi.org/10.1890/08-1823.1 was used to infer abundance differences between pre-Blob and Blob in different taxonomic groups (OTUs).
Because The Blob was located in the upper 0-100m when it first appeared at the end of 2013, and subsequently deepened down to 200m in 2015 we ran two indicator analyses on the 10-100m samples and the 150-200m, respectively. 

Resulting indicators was filtered and OTUs with stat > 0.7 and p-value < 0.05 was retained for further analysis.
```{r, echo = FALSE}
library(indicspecies) # v. 1.7.6
library(tidyverse)
prop.otu <- read.csv("otu_prop_w_metadata_phyloseq5.csv", header = TRUE)
## Indicator species: Blob vs normal years for Surface Samples
surf.samples <- prop.otu %>% filter(SampleDepth < 110) #subsetting data only keeping surface samples (0-100m)
indic.BS <- multipatt(surf.samples[,37:1346], surf.samples[,4], func = "IndVal.g", duleg = TRUE, control = how(nperm=999))
  # duleg = TRUE the function does not take site group combinations into considerations.
  # we set the assocation function to IndVal.g assuming all site groups have equal ecological variability
  # Indicator value index is the product of component 'A' and 'B'.
  # 'A' is the probability that the surveyed site belongs to the target site group - the predictive value of the species as an indicator to the site group.
  # 'B' is the probability of finding the species in sites belonging to the site group.
  # to report 'A' and 'B' use indvalcomp = TRUE
indic.BS2 <- capture.output(summary(indic.BS, indvalcomp = TRUE)) # capture content of the summary
write.table(indic.BS2,file("IndicatorAnalysis_Blob_Surface_Apr_2019.txt")) # save summary content as .csv

# Indicator species: Blob vs normal years for Lower Surface Samples
Lsurf.samples <- prop.otu %>% filter( SampleDepth > 130) %>%
  filter(SampleDepth < 210) #subsetting data to 110-200m samples

indic.BLS <- multipatt(Lsurf.samples[,37:1346], Lsurf.samples[,4], func = "IndVal.g", duleg = TRUE, control = how(nperm=999))
  # duleg = TRUE the function does not take site group combinations into considerations.
  # we set the assocation function to IndVal.g assuming all site groups have equal ecological variability
  # Indicator value index is the product of component 'A' and 'B'.
  # 'A' is the probability that the surveyed site belongs to the target site group - the predictive value of the species as an indicator to the site group.
  # 'B' is the probability of finding the species in sites belonging to the site group.
  # to report 'A' and 'B' use indvalcomp = TRUE
indic.BLS2 <- capture.output(summary(indic.BLS, indvalcomp = TRUE)) # capture content of the summary
write.table(indic.BLS2,file("IndicatorAnalysis_Blob_Lower_Surface_Nov_2019.txt")) # save summary content as .csv
```

### Network analysis
In order to not exclude conditionally rare taxa a co-occurrence network analysis was applied to further filter the indicator OTUs by looking at their connectivity within the community.

The CoNet v. 1.1.1.beta (Faust and Raes, 2016 (https://doi.org/10.12688/f1000research.9050.2)) application within Ctyoscape v. 3.7.1 (Shannon et al., 2003, http://www.genome.org/cgi/doi/10.1101/gr.1239303.) was applied to the two subsets of data used for the indicator analyses (10-100m and 150-200m). Only positive correlations was retained and indicator OTUs  >= 3 connections was retained for discussion.

## Figure 1 Sampling location and environmental parameters

Panel B historical temperature in the top 400m at OSP.
Panel C average temperature for 10-100 and 150-200m, respectively, for the period covering our samples. These will be used as sparklines for subsequent figures.

### Figure 1A
Geographic location of the Blob in February 2014. Made by T. Ross, using Matlab.

### Figure 1B
Historical temperature in the top 400m at OSP. Made by T. Ross, using Matlab.

### Figure 1C
Average temperature sparklines between June 2010 to February 2016, for 10-100m and 150-200m.

### Figure 1D
Average chlorophyll concentration for the upper 60m. Sample depth varies between years depending on environmental conditions.
```{r, echo=F}
library(tidyverse)
library(ggplot2)
library(gridExtra)
library(cowplot)

IOS.data <- read.csv("./T10-100m.csv", header = TRUE, stringsAsFactors = FALSE)
# split date stamp for taking averages
IOS.dat1 <- IOS.data %>% separate(., Date, c("year", "month","day","hours","min")) %>% unite(., datelabel, c(year,month), sep = "_", remove = FALSE) %>% group_by(datelabel) %>% dplyr::summarize(mean = mean(mean.T.10.100m)) 
sparkline_upper <- IOS.dat1[1:18,]
IOS.data2 <- read.csv("./T150-200m.csv", header = TRUE, stringsAsFactors = FALSE)
# split date stamp for taking averages
IOS.dat3 <- IOS.data2 %>% separate(., Date, c("year", "month","day","hours","min")) %>% unite(., datelabel, c(year,month), sep = "_", remove = FALSE) %>% group_by(datelabel) %>% dplyr::summarize(mean = mean(mean.T.150.200m)) 
sparkline_lower <- IOS.dat3[1:18,]

p.spark <- ggplot() + 
  geom_line(data = sparkline_upper, aes(x = datelabel, y = mean, group = 1)) +
  geom_line(data = sparkline_lower, aes(x = datelabel, y = mean, group = 1 )) +
  ylim(2,10) +
  xlab("Dates") +
  ylab("temperature [ °C]") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
# Chlorophyll
TotChl <- pigment %>% separate(., Cruisetrans, c("station", "year", "month","day", "depth")) %>% unite(., datelabel, c(year,month), sep = "_", remove = FALSE) %>% group_by(datelabel) %>% summarise(all.chl = mean(Chlorophyll.mg.m3))
TotChl.adjust <- sparkline_upper %>% left_join(TotChl, by = "datelabel") # adding the one datapoint missing in the chl data
p.sparkC <- ggplot() + 
  geom_line(data = TotChl.adjust, aes(x = datelabel, y = all.chl, group = 1)) +
  xlab("Dates") +
  ylab("Chl [mg m3]") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Combined panel
panelCD <- plot_grid(p.spark, p.sparkC, ncol=1)
```

## Supplementary figure S1 - Environmetal parameters
Depth profiles of measured environmental parameters at OSP.
These are collected by the IOS, Sidney, BC, Canada. Raw data available from https://waterproperties.ca/linep/ 
```{r, echo = FALSE}
library(tidyverse)
library(ggplot2)
## Water column profiles of environmental parameters
env.ord <- meta_dat %>% mutate(mdepth = fct_relevel(mdepth, # order samples according to depth
                              "B-10 m",
                              "4000 m",
                              "3000 m",
                              "2000 m",
                              "1500 m",
                              "1250 m",
                              "1000 m",
                              "0800 m",
                              "0600 m",
                              "0400 m",
                              "0300 m",
                              "0200 m",
                              "0150 m",
                              "0125 m",
                              "0100 m",
                              "0050 m",
                              "0025 m",
                              "0010 m")) %>% arrange(mdepth)

## SALINITY ##
Salt <- ggplot(env.ord, aes(x=mdepth, y=salnCTD)) + geom_boxplot()
Salt + coord_flip()
ggsave(file = "Salt_boxplot_Mar_2019.eps")
## TEMPERATURE ##
Tem <- ggplot(env.ord, aes(x=mdepth, y=tempCTD)) + geom_boxplot()
Tem + coord_flip()
ggsave(file = "Tem_boxplot_Mar_2019.eps")
## OXYGEN ##
Oxy <- ggplot(env.ord, aes(x=mdepth, y=O2)) + geom_boxplot()
Oxy + coord_flip()
ggsave(file = "Oxy_boxplot_Mar_2019.eps")
## Nitrite and Nitrate ## 
NO2NO3 <- ggplot(env.ord, aes(x=mdepth, y=NO2NO3)) + geom_boxplot()
NO2NO3 + coord_flip()
ggsave(file = "NO2NO3_boxplot_Mar_2019.eps")
## Phosphate ##
PO4 <- ggplot(env.ord, aes(x=mdepth, y=PO4)) + geom_boxplot()
PO4 + coord_flip()
ggsave(file = "PO4_boxplot_Mar_2019.eps")
## Silica ## 
Si <- ggplot(env.ord, aes(x=mdepth, y=SiO4)) + geom_boxplot()
Si + coord_flip()
ggsave(file = "SiO4_boxplot_Mar_2019.eps")
## Fluorescence from CTD (SP) ##
Flu <- ggplot(env.ord, aes(x=mdepth, y=FluorSPCTD)) + geom_boxplot()
Flu + coord_flip()
ggsave(file = "FluorSPCTD_boxplot_Mar_2019.eps")
```

## Figure 2 - T-S and Beta-diversity

Temperature-Salinity (T-S) relationship plot showing the T-S profile from CTD data, overlayed with our sample data.
```{r}
library(R.matlab)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(cowplot)

ctddata <- readMat("./StationP26_forSachia.mat")
ctddates <- read.csv("./StationP26_CTDdates_forSachia_w_header.csv", header = TRUE)
ctd.dates <- unite_(ctddates, "yyyymm", c("year", "month"))
ctd.sal <- ctddata$Salinity
ctd.temp <- ctddata$Temperature
ctd.depth <- ctddata$dep
ctd.sal <- as.data.frame(ctddata$Salinity)
ctd.temp <- as.data.frame(ctddata$Temperature)
ctd.utc <- as.data.frame(t(ctddata$t.utc))
colnames(ctd.sal) <- ctd.depth # rename columns using depth
colnames(ctd.temp) <- ctd.depth # rename columns using depth
ctd.sal <- cbind(ctd.utc,ctd.dates, ctd.sal) 
ctd.temp <- cbind(ctd.dates,ctd.temp) 
# Format ctd data into a data.frame so ggplot can take it
ctd.salL <- gather(ctd.sal, depth, salinity,6:865, factor_key=TRUE)
ctd.tempL <- gather(ctd.temp, depth, temperature, 5:864, factor_key = TRUE)
# Combine salinity and temp into one table
ctd.all <- as.data.frame(cbind(ctd.salL, ctd.tempL$temperature), StringAsFactors = FALSE)

# Plot ctd data
ctd.plot <- ggplot(ctd.all, aes(x=salinity, y=ctd.tempL$temperature), ylab = "Tempeature (°C)", xlab = "Salinity") +       
    geom_point() 

oursamples <- ggplot(metadata, aes(x = salnCTD, y = tempCTD, colour = metadata$Depths), ylab = "Tempeature (°C)", xlab = "Salinity") +       
    geom_point() + 
    scale_colour_discrete(drop=TRUE,
        limits = levels(metadata$Depths)) + scale_colour_manual(values = DepthColSET) +
  theme(legend.position = "none")

p1 <- plot_grid(ctd.plot, oursamples,  align = "v", nrow = 2, labels = c("A", "B"))
# Export figure from plot window
#for legend
oursamples2 <- ggplot(metadata, aes(x = salnCTD, y = tempCTD, colour = metadata$Depths), ylab = "Tempeature (°C)", xlab = "Salinity") +       
    geom_point() + 
    scale_colour_discrete(drop=TRUE,
        limits = levels(metadata$Depths)) + scale_colour_manual(values = DepthColSET)
```

NMDS plot showing community composition similarity
```{r, cache.comments= FALSE, warning=FALSE, echo=FALSE}
library(vegan) # v. 2.5-4
library(permute)
library(lattice)

otu_table_vst1 <- read.csv("vst_otu_table.csv", header = TRUE, row.names = 1)
otu_table_vst <- read.csv("vst_otu_table_transposed.csv", header = TRUE, row.names = 1)
set.seed(1)
otu_vst <- vegdist(t(otu_table_vst1), method="bray")
MDS.vst <- metaMDS(otu_vst, trace = FALSE, k = 2, trymax = 100 )
plot(MDS.vst, type = "t")
stressplot(MDS.vst)
ggsave(file = "nmds_stressplot_vst_table_Mar_2019.eps")

# color nmds plot and make it nicer using ggplot
data.scores <- as.data.frame(scores(MDS.vst))
data.scores$site <- rownames(data.scores)
grp.depth <- factor(meta_dat$mdepth) # the grouping we want to color our mds plot after
grp.season <- factor(meta_dat$Months) # this grouping is for Martin's suggestion of open and closed circles

data.scores$grp.depth <- grp.depth

NMDS.all <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, colour = grp.depth), size = 3) +
  scale_colour_manual(values = DepthCol2)
# Saved as NMDS_all_samples_Mar_2019.eps
  # dim: 900 x 600

###############################################
NMDS.allMartin <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, colour = grp.depth, shape=as.factor(grp.season)), size = 3) +
  scale_colour_manual(values = DepthCol2) + scale_shape_manual(values = c(21,19,21))
# Saved as NMDS_all_samples_Mar_2020.eps
  # dim: 900 x 600
##############################################

# ggplot's geom_point will automatically not plot NAs, we therefore change them to 0
NMDS.base <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Chlorophyll_avg))

meta_dat1 <- meta_dat1 %>% mutate(Chlorophyll_norm = log(Chlorophyll_avg))

meta_dat1$Chlorophyll_avg[is.na(meta_dat1$Chlorophyll_avg)] <- 0
NMDS.TChl <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Chlorophyll_avg)) +
  scale_size(range = c(1,10))

### Log transformed Chl data
meta_dat1 <- meta_dat1 %>% mutate(Chlorophyll_norm = log(Chlorophyll_avg))
NMDS.logChl <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Chlorophyll_norm)) +
  scale_size(range = c(1,10))

### Nitrogen
NMDS.nitro <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$NO2NO3)) +
  scale_size(range = c(0.5,8))

### Temperature
NMDS.temp <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$tempCTD)) +
  scale_size(1,10) + guides(size = guide_legend(order =3))

### Blob year
NMDS.blob <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, colour = meta_dat1$BlobYear, size = 3)) +
  scale_colour_manual(values = Blob.colors)
```

## Supplementary figure S2 - Beta-diversity
NMDS plot showing community composition similarity with point size scaled with the pigment concentration of eight different phytoplankton groups.
```{r, cache.comments= FALSE, warning=FALSE, echo=FALSE}
library(vegan)
library(permute)
library(lattice)
otu_table_vst <- read.csv("vst_otu_table_transposed.csv", header = TRUE)
otu_table_vst1 <- read.csv("vst_otu_table.csv", header = TRUE, row.names = 1)
meta_dat1 <- read.table("metadata_with_pigments.tsv", header = TRUE, row.names = 1, sep="\t")

set.seed(1)
# Need to transpose otu table to work with vegan functions
otu_vst <- vegdist(t(otu_table_vst1), method="bray")
MDS.vst <- metaMDS(otu_vst, trace = FALSE, k = 2, trymax = 100 )
# color nmds plot and make it nicer using ggplot
data.scores <- as.data.frame(scores(MDS.vst))
data.scores$site <- rownames(data.scores)

# Prasinophytes
meta_dat1$Prasinophytes_avg[is.na(meta_dat1$Prasinophytes_avg)] <- 0
NMDS.Pra <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Prasinophytes_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Pelagophytes
meta_dat1$Pelagophytes_avg[is.na(meta_dat1$Pelagophytes_avg)] <- 0
NMDS.Pel <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Pelagophytes_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Haptophytes
meta_dat1$Haptophytes_avg[is.na(meta_dat1$Haptophytes_avg)] <- 0
NMDS.Hap <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Haptophytes_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Dinoflage1
meta_dat1$Dinoflage1_avg[is.na(meta_dat1$Dinoflage1_avg)] <- 0
NMDS.Din <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Dinoflage1_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Diatom2
meta_dat1$Diatom2_avg[is.na(meta_dat1$Diatom2_avg)] <- 0
NMDS.Dia <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Diatom2_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Cyanobacteria
meta_dat1$Cyanobacteria_avg[is.na(meta_dat1$Cyanobacteria_avg)] <- 0
NMDS.Cya <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Cyanobacteria_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Cryptophytes
meta_dat1$Cryptophytes_avg[is.na(meta_dat1$Cryptophytes_avg)] <- 0
NMDS.Cry <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Cryptophytes_avg)) +
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
# Chlorophytes
meta_dat1$Chlorophytes_avg[is.na(meta_dat1$Chlorophytes_avg)] <- 0
NMDS.Chlo <- ggplot() +
  geom_point(data = data.scores, aes(x = NMDS1, y = NMDS2, size = meta_dat1$Chlorophytes_avg) )+
  scale_size(range = c(1,10)) + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(), axis.line = element_line(colour = "black"))
```

## Figure 3 - Heatmap of phyla abundance and sample clustering

### Hierarchical clustering of samples
First we test the data to see what the optimal number of clusters are for explaining our samples. We use the Calinski method and also kmeans which is shown as an elbow plot.
```{r,cache.comments = FALSE, echo = FALSE, warning = FALSE}
library(vegan)
library(lattice)
library(factoextra)

# The Calinski plot from vegan
fit <- cascadeKM(scale(t(otu_table_vst1), center = TRUE, scale = TRUE), 1, 20, iter = 1000)
plot(fit, sortg = TRUE, grpmts.plot = TRUE) # execute in console
calinski.best <- as.numeric(which.max(fit$results[2,]))
cat("Calinski criterion optimal numbers of clusters", calinski.best, "/n")
## Results
  # Calinski criterion optimal numbers of clusters 4/n

# Kmeans and elbow plot
otu_vst <- as.matrix(otu_vst) # we reuse the vegdist object calculated above
fviz_nbclust(otu_vst, kmeans, method = "wss") +
  geom_vline(xintercept = 4, linetype = 2) +
  labs(subtitle = "elbow plot")
ggsave(file = "Elbow_plot_Mar_2019.eps")

# Cutting tree using 4 clusters as suggested to be most optimal 
hc.bc <- hclust(as.dist(otu_vst), method = "average")
# hclust needs to be told if you use a prevously calculated dist matrix.
# In this case we calculated a distance matrix (using vegan) earlier and called it "otu_vst"
plot(hc.bc) # execute in console
rect.hclust(hc.bc, k = 4, border = 2:6) # color boxes indicating the four clusters
cut.bc4 <- cutree(hc.bc, k = 4) 
write.table(cut.bc4,file("sample_clustering_4_Mar_2019.txt")) # list of sample names and their cluster group

### color dendrogram by temperature
### cant figure out if a continueous variable can be used, all examples are for categories
order2 <- read.csv("./hclust_sample_order.csv", header = F)
tem <- inner_join(order2, metadata, by = c("V1"="SampleID")) %>% select(., "V1","tempCTD")
tem$V1.ordered <- factor(tem$V1, levels = tem$V1 )
p.tem <- ggplot(tem, aes(x=V1.ordered, y=V1.ordered)) + geom_point(aes(color=tempCTD)) + scale_color_gradient(low="#0000FF", high="#FF0404")
```

### Heatmap to accompany the dendrogram
We aggregated abundance at the level of Phylum. Since the dendrogram and heatmap will be at two different levels of resolution, we will not ask the heatmap to cluster our samples. instead we pull the clustering order from the hclust analysis already performed on the data (see above). For the heatmap we will use the function from Phyloseq.
```{r, echo=F}

#extract order from all OTU cluster analysis NOT USED
order<-hc.bc$labels[c(hc.bc$order)] #if you just do clust$order you get back numbers instead of labels. so you have to do clust$labels[c(clust$order)]
order2 <- read.csv("./hclust_sample_order.csv")
order2.1 <- order2[,1] # ended using this as we changed the hclust tree so I manually arranged the list

#transform abundance to proportions
P26.P = transform_sample_counts(P26, function(x)x/sum(x))

# Agglomerate abundances by Phyla
P26.phyla = tax_glom(P26.P, "Phylum_mod") 

taxOrder2 <- names(sort(taxa_sums(P26.phyla), decreasing = F))
#make a heatmap in phyloseq based on the agglomerated/collapsed by phylum OTU table, but using the order from the first cluster analysis (instead of the one plot_heatmap would otherwise do) and label the taxa by phylum
plot_heatmap(P26.phyla, sample.order = order2.1, taxa.order = taxOrder2,taxa.label = "Phylum_mod", max.label = 300, high = "#000000", low="#D0D0D0", na.value="#FFFFFF")
```

## Supplementary figure S3 - Alpha-diversity

All diversity metrics are based on a OTU table with singletons, mitochondrial, and chloroplast reads removed. (smallest OTU = 2 reads)
```{r, cache.comments = FALSE, echo = FALSE, warning = FALSE}
# Shannon diversity
shannonplot = plot_richness(P26, x="mdepth", measures=c("shannon")) + 
  coord_flip() + 
  theme_bw() + 
  scale_x_discrete(limits =rev(levels(get_variable(P26,"mdepth")))) +
  geom_boxplot( color= I("#000000"), outlier.color = "#ff0000") +
  theme(legend.position = "none",axis.text=element_text(size=12), strip.background = element_blank(), strip.text.x = element_blank()) +
  ylab("Shannon Diversity Index") +
  xlab("Depth (m)")
```

Plotting a-diversity over time at the different sampling depths.
This means we will ask Phyloseq to calculate a a-div value for each sample so we can plot it, this is what the "split" argument does below.
```{r, echo = FALSE}
# Calculate a diversity value for each sample
P26_adiv_S <- estimate_richness(P26, split = TRUE, measure = "Shannon")
P26_adiv_C <- estimate_richness(P26, split = TRUE, measure = "Chao1")
P26_adiv_Si <- estimate_richness(P26, split = TRUE, measure = "Simpson")

library(data.table)
setDT(P26_adiv_S, keep.rownames = TRUE) #copy rownames to a new column and I don't know better.
setDT(P26_adiv_C, keep.rownames = TRUE)
setDT(P26_adiv_Si, keep.rownames = TRUE)
setDT(meta_dat, keep.rownames = TRUE)

#stitch on the info on Blob and depth
metadata.adiv <- meta_dat %>% inner_join(P26_adiv_S, by = c("rn" = "rn")) %>% 
  inner_join(P26_adiv_C, by = c("rn" = "rn")) %>% 
  inner_join(P26_adiv_Si, by = c("rn" = "rn"))
```

```{r, cache.comments = FALSE, echo = FALSE, warning = FALSE}
cruiseorderlabels = c("J10", "A10", "F11", "J11", "A11", "F12", "J12", "A12", "F13", "J13", "A13", "F14", "J14", "A14", "F15", "J15", "A15", "F16")
# Alpha-diversity using the Shannon Index, with Phyloseq values

shannon_time = ggplot(metadata.adiv, aes(CruiseOrder, Shannon)) +
  geom_line(aes(group=mdepth), size = .5) + 
  scale_x_continuous( breaks = c(1:18),labels = cruiseorderlabels ) +
  facet_grid(mdepth~.)
```

## Figure 4 and supplementary figure S5 - Taxonomy distributions

Boxplot of all phyla present in the water column, with phyla arranged by size.
```{r, echo=F}
# run chunks which create the phyloseq object and the depth colorcoding, they are at the top of this document
# run first chunk of Heatmap chunks, above this, where the OTU table is aggregated into phyla following the customized phylum list

P26.plong <- P26.phyla %>% psmelt() %>% mutate(Phylum_mod = fct_relevel(Phylum_mod,"Alphaproteobacteria",
                              "Thaumarchaeota",
                              "Gammaproteobacteria",
                              "Deltaproteobacteria",
                              "Marinimicrobia (SAR406 clade)",
                              "Bacteroidetes",
                              "Euryarchaeota",
                              "Nitrospinae",
                              "Planctomycetes",
                              "Chloroflexi",
                              "Actinobacteria",
                              "Verrucomicrobia",
                              "Cyanobacteria",
                              "Acidobacteria",
                              "Gemmatimonadetes",
                              "Nanoarchaeaeota",
                              "Bacteria",
                              "PAUC34f",
                              "AncK6",
                              "Dadabacteria",
                              "Proteobacteria",
                              "Margulisbacteria",
                              "Archaea",
                              "Patescibacteria",
                              "Kiritimatiellaeota",
                              "Lentisphaerae",
                              "Hydrogenedentes",
                              "Fibrobacteres",
                              "Firmicutes")) %>% arrange(Phylum_mod)
# Boxplot of all Phyla
p.phyla <- ggplot(P26.plong, aes(x=Phylum_mod, y=Abundance)) + geom_boxplot(outlier.size = -1) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Boxplot of 1-10
box.fig3 <- P26.plong %>% filter(Phylum_mod == c("Alphaproteobacteria",
                              "Thaumarchaeota",
                              "Gammaproteobacteria",
                              "Deltaproteobacteria",
                              "Marinimicrobia (SAR406 clade)",
                              "Bacteroidetes",
                              "Euryarchaeota",
                              "Nitrospinae",
                              "Planctomycetes",
                              "Chloroflexi"))
pbox.fig3 <- ggplot(box.fig3, aes(x=Phylum_mod, y=Abundance)) + geom_boxplot(outlier.size = 1) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Boxplot of 11-20
box.S5a <- P26.plong %>% filter(Phylum_mod == c("Actinobacteria",
                              "Verrucomicrobia",
                              "Cyanobacteria",
                              "Acidobacteria",
                              "Gemmatimonadetes",
                              "Nanoarchaeaeota",
                              "Bacteria",
                              "PAUC34f",
                              "AncK6",
                              "Dadabacteria"))
pbox.S5a <- ggplot(box.S5a, aes(x=Phylum_mod, y=Abundance)) + geom_boxplot(outlier.size = 1) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0, 0.15)

# Boxplot of 21-29
box.S5b <- P26.plong %>% filter(Phylum_mod == c("Proteobacteria",
                              "Margulisbacteria",
                              "Archaea",
                              "Patescibacteria",
                              "Kiritimatiellaeota",
                              "Lentisphaerae",
                              "Hydrogenedentes",
                              "Fibrobacteres",
                              "Firmicutes"))
pbox.S5b <- ggplot(box.S5b, aes(x=Phylum_mod, y=Abundance)) + geom_boxplot(outlier.size = 1) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylim(0,0.006)
```

Average abundance and standard deviation with depth for each phyla.
```{r, echo=FALSE}
# Individial phyla
alpha <- P26.plong %>% filter(Phylum_mod == 'Alphaproteobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.alpha <- ggplot(alpha, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(alpha$mdepth))) + coord_flip()

thau <- P26.plong %>% filter(Phylum_mod == 'Thaumarchaeota') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.thau <- ggplot(thau, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(thau$mdepth))) + coord_flip() 

gamma <- P26.plong %>% filter(Phylum_mod == 'Gammaproteobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.gamma <- ggplot(gamma, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(gamma$mdepth))) + coord_flip() 

delta <- P26.plong %>% filter(Phylum_mod == 'Deltaproteobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.delta <- ggplot(delta, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(delta$mdepth))) + coord_flip() 

sar <- P26.plong %>% filter(Phylum_mod == 'Marinimicrobia (SAR406 clade)') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.sar <- ggplot(sar, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(sar$mdepth))) + coord_flip()

bac <- P26.plong %>% filter(Phylum_mod == 'Bacteroidetes') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.bac <- ggplot(bac, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(bac$mdepth))) + coord_flip() 

eur <- P26.plong %>% filter(Phylum_mod == 'Euryarchaeota') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.eur <- ggplot(eur, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(eur$mdepth))) + coord_flip()

nit <- P26.plong %>% filter(Phylum_mod == 'Nitrospinae') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.nit <- ggplot(nit, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(nit$mdepth))) + coord_flip()

pla <- P26.plong %>% filter(Phylum_mod == 'Planctomycetes') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.pla <- ggplot(pla, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(pla$mdepth))) + coord_flip()

chx <- P26.plong %>% filter(Phylum_mod == 'Chloroflexi') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.chx <- ggplot(chx, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(chx$mdepth))) + coord_flip()

act <- P26.plong %>% filter(Phylum_mod == 'Actinobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.act <- ggplot(act, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(act$mdepth))) + coord_flip()

ver <- P26.plong %>% filter(Phylum_mod == 'Verrucomicrobia') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.ver <- ggplot(ver, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(ver$mdepth))) + coord_flip() 

cya <- P26.plong %>% filter(Phylum_mod == 'Cyanobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.cya <- ggplot(cya, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(cya$mdepth))) + coord_flip() 

aci <- P26.plong %>% filter(Phylum_mod == 'Acidobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.aci <- ggplot(aci, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(aci$mdepth))) + coord_flip() 

gem <- P26.plong %>% filter(Phylum_mod == 'Gemmatimonadetes') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.gem <- ggplot(gem, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(gem$mdepth))) + coord_flip()

nano <- P26.plong %>% filter(Phylum_mod == 'Nanoarchaeaeota') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.nano <- ggplot(nano, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(nano$mdepth))) + coord_flip()

bact <- P26.plong %>% filter(Phylum_mod == 'Bacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.bact <- ggplot(bact, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(bact$mdepth))) + coord_flip() 

pau <- P26.plong %>% filter(Phylum_mod == 'PAUC34f') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.pau <- ggplot(pau, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(pau$mdepth))) + coord_flip() 

anc <- P26.plong %>% filter(Phylum_mod == 'AncK6') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.anc <- ggplot(anc, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(anc$mdepth))) + coord_flip()

dad <- P26.plong %>% filter(Phylum_mod == 'Dadabacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.dad <- ggplot(dad, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(dad$mdepth))) + coord_flip() 

pro <- P26.plong %>% filter(Phylum_mod == 'Proteobacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.pro <- ggplot(pro, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(pro$mdepth))) + coord_flip()

mar <- P26.plong %>% filter(Phylum_mod == 'Margulisbacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.mar <- ggplot(mar, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(mar$mdepth))) + coord_flip() 

arc <- P26.plong %>% filter(Phylum_mod == 'Archaea') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.arc <- ggplot(arc, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(arc$mdepth))) + coord_flip()

pat <- P26.plong %>% filter(Phylum_mod == 'Patescibacteria') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.pat <- ggplot(pat, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(pat$mdepth))) + coord_flip()

kir <- P26.plong %>% filter(Phylum_mod == 'Kiritimatiellaeota') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.kir <- ggplot(kir, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(kir$mdepth))) + coord_flip()

len <- P26.plong %>% filter(Phylum_mod == 'Lentisphaerae') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.len <- ggplot(len, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(len$mdepth))) + coord_flip() 

hyd <- P26.plong %>% filter(Phylum_mod == 'Hydrogenedentes') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.hyd <- ggplot(hyd, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(hyd$mdepth))) + coord_flip()

fib <- P26.plong %>% filter(Phylum_mod == 'Fibrobacteres') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.fib <- ggplot(fib, aes(x=mdepth, y = meana, group = 1)) + geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + geom_line() + scale_x_discrete(limits = rev(levels(fib$mdepth))) + coord_flip()

fir <- P26.plong %>% filter(Phylum_mod == 'Firmicutes') %>% group_by(mdepth) %>% summarise(meana=mean(Abundance), sd = sd(Abundance), maxa = max(Abundance), mina=min(Abundance)) %>% drop_na(sd)
p.fir <- ggplot(fir, aes(x=mdepth, y = meana, group = 1)) + 
  geom_ribbon(aes(ymin=meana-sd, ymax = meana+sd)) + 
  geom_line() + 
  scale_x_discrete(limits = rev(levels(fir$mdepth))) + 
  coord_flip() 

library(gridExtra)
library(cowplot)

p.fig3 <- plot_grid(p.alpha, p.thau, p.gamma, p.delta, p.sar, p.bac, p.eur, p.nit, p.pla, p.chx,align = "v", nrow = 1 , labels = c("Alphaproteobacteria",
                              "Thaumarchaeota",
                              "Gammaproteobacteria",
                              "Deltaproteobacteria",
                              "Marinimicrobia (SAR406 clade)",
                              "Bacteroidetes",
                              "Euryarchaeota",
                              "Nitrospinae",
                              "Planctomycetes",
                              "Chloroflexi"))
p.S5a <- plot_grid(p.act, p.ver, p.cya, p.aci, p.gem, p.nano, p.bact, p.pau, p.anc, p.dad,
                   align = "v", nrow = 1 , 
                   labels = c("Actinobacteria",
                              "Verrucomicrobia",
                              "Cyanobacteria",
                              "Acidobacteria",
                              "Gemmatimonadetes",
                              "Nanoarchaeaeota",
                              "Unassigned Bacteria",
                              "PAUC34f",
                              "AncK6",
                              "Dadabacteria"))
p.S5b <- plot_grid(p.pro, p.mar, p.arc, p.pat, p.kir, p.len, p.hyd, p.fib, p.fir,
                   align = "v", nrow = 1 , 
                   labels = c("Unassigned Proteobacteria",
                              "Margulisbacteria",
                              "Unassigned Archaea",
                              "Patescibacteria",
                              "Kiritimatiellaeota",
                              "Lentisphaerae",
                              "Hydrogenedentes",
                              "Fibrobacteres",
                              "Firmicutes"))
    
combo1 <- plot_grid(pbox.fig3, p.fig3, ncol=1, rel_heights = c(1.5,2.5))
combo2 <- plot_grid(pbox.S5a, p.S5a, ncol=1, rel_heights = c(1.5,2.5))
combo3 <- plot_grid(pbox.S5b, p.S5b, ncol=1, rel_heights = c(1.5,2,5))
```

## Supplementary figure 4 - Heatmap of indicator abundances

### S4A
Heatmap for all indicator OTUs in upper surface sample (10-100m).
No = pre-Blob indicators
Yes = Blob indicators
```{r, echo=F}
library(ComplexHeatmap)
library(circlize) #for making custom colors 



BS_otu_table <- read.csv("./Blob_Surface_indicators_abund_tax_Sep_2019.csv", header = T, row.names = 1)
#create factor for taking average of OTU abundance across the 10-100m water column
BSavg <- BS_otu_table %>% gather(., SampleID, abundance, P26.2010.06.10m:P26.2016.02.100m) %>%
  separate(., SampleID, c("station", "year", "month", "depth")) %>% 
  unite(.,date, c(year, month), sep = "_", remove = F) %>%
  group_by(OTUID.1,date) %>%
  dplyr::summarize(average = mean(abundance, na.rm = T)) %>%
  spread(., date, average)
BS.order <- BS_otu_table$OTUID.1
BSavg2 <- inner_join(BSavg, BS_otu_table, by = "OTUID.1") %>% arrange(factor(OTUID.1, levels = BS.order))
BSmat <- log(BSavg2[,2:19]) # log transform as there are huge differences in the OTUs proportion sizes.

is.na(BSmat) <- sapply(BSmat, is.infinite) #0 cannot be log-trans and are "Inf" we want to convert these to NaN
BSmat[is.na(BSmat)] <- NaN # see above
row.names(BSmat) <- BSavg2$OTUID.1
f1 = colorRamp2(seq(min(BSmat, na.rm = T), max(BSmat, na.rm = T), length = 2), c("#D0D0D0","#000000" ))
f2 = colorRamp2(c(-1.59, -13.81551),c("#000000","#D0D0D0") )
f3 = colorRamp2(c(-1, -14),c("#000000","#D0D0D0") )
blob.grp <- data.frame(BSavg2$Group)
rowann.blob <- HeatmapAnnotation(df = blob.grp, which = "row")
phylum_mod <- data.frame(BSavg2$Phylum_mod)
rowann.phyl <- HeatmapAnnotation(df = phylum_mod, which = "row")
### Heatmap
Heatmap(BSmat, cluster_columns = F, cluster_rows = F, col = f3, na_col = "#FFFFFF", heatmap_legend_param = list(at = c(-1.59,-4.6, -9.21, -13.82), labels = c("0.203","0.01","0.0001", "0.000001"))) + rowann.blob + rowann.phyl 

# Found max and min value before log
leg2 = Legend(at = c(-1.5, -5, -10, -13.8), col_fun = f2, title = "custom")
draw(leg2, x = unit(1, "cm"), y = unit(1, "cm"), just = c("left", "bottom"))
### The scale is log-transformed in order for better visual representation the values are not intuitive, they are therefore replaced with the abundance values before log transformation.
colMax <- function(data) sapply(data, max, na.rm=T)
colMax(BSavg2[,2:19])
colMin <- function(data) sapply(data, min, na.rm=T)
colMin(BSmat) # since 
# Largest value = 0.20315336 (-1.593794)
# smallest value = 0.000001 (-13.81551)
```

Heatmap for all indicator OTUs in lower surface samples (150-200m)
November 2019: redid analysis
```{r, echo=F}
library(ComplexHeatmap)
library(circlize) #for making custom colors 

BLS_otu_table <- read.csv("./Blob_Lower_Surface_indicators_propabund_tax_Nov_2019.csv", header = T, row.names = 1)
#create factor for taking average of OTU abundance across the 150-200m water column
BLSavg <- BLS_otu_table %>% gather(., SampleID, abundance, P26.2010.06.150m:P26.2016.02.200m) %>%
  separate(., SampleID, c("station", "year", "month", "depth")) %>% 
  unite(.,date, c(year, month), sep = "_", remove = F) %>%
  group_by(OTUID.1,date) %>%
  dplyr::summarize(average = mean(abundance, na.rm = T)) %>%
  spread(., date, average)
BLS.order <- BLS_otu_table$OTUID.1
BLSavg2 <- inner_join(BLSavg, BLS_otu_table, by = "OTUID.1") %>% arrange(factor(OTUID.1, levels = BLS.order))
BLSmat <- log(BLSavg2[,2:17]) # log transform as there are huge differences in the OTUs proportion sizes.

is.na(BLSmat) <- sapply(BLSmat, is.infinite) #0 cannot be log-trans and are "Inf" we want to convert these to NaN
BLSmat[is.na(BLSmat)] <- NaN # see above
row.names(BLSmat) <- BLSavg2$OTUID.1
f5 = colorRamp2(seq(min(BLSmat, na.rm = T), max(BLSmat, na.rm = T), length = 2), c("#D0D0D0","#000000" ))
f6 = colorRamp2(c(-2.733318, -13.03389),c("#000000","#D0D0D0") )
f7 = colorRamp2(c(-1, -14),c("#000000","#D0D0D0") )

Lblob.grp <- data.frame(BLSavg2$Group)
rowann.Lblob <- HeatmapAnnotation(df = Lblob.grp, which = "row")
phylum_modL <- data.frame(BLSavg2$Phylum_mod)
rowann.Lphyl <- HeatmapAnnotation(df = phylum_modL, which = "row")
### Heatmap
Heatmap(BLSmat, cluster_columns = F, cluster_rows = F, col = f6, na_col = "#FFFFFF", heatmap_legend_param = list(at = c(-2.73,-4.6, -9.21, -13.03), labels = c("0.065","0.01","0.0001", "0.000002"))) + rowann.Lblob + rowann.Lphyl 

# Find max and min value before log
### The scale is log-transformed in order for better visual representation the values are not intuitive, they are therefore replaced with the abundance values before log transformation.
colMax(BLSavg2[,2:17])
colMin <- function(data) sapply(data, min, na.rm=T)
colMin(BLSmat) 
```

## Figure S6 workflow diagram
Created by Ryan McLaughlin in XXX

